src/components/tasks/task_detail_menu_component.tsx

```
@@ -0,0 +1,68 @@
+import React from 'react';
+import * as R from 'ramda';
+import { Trans } from '@lingui/react';
+import { Icon, Text, Button } from 'native-base';
+import { applicationStyles, textStyles, colors } from '../../application/styles';
+import { TaskStateButton } from './task_states';
+import { FloatingActionButtonsComponent } from '../floating_action_buttons/floating_action_buttons_component';
+
+export interface TaskDetailMenuProps {
+    readonly buttons: ReadonlyArray<TaskStateButton>;
+}
+
+export interface TaskDetailMenuActions {
+    readonly addButtonOnPress: () => void;
+    readonly removeButtonOnPress: () => void;
+    readonly doneButtonOnPress: () => void;
+    readonly notDoneButtonOnPress: () => void;
+}
+
+type Props = TaskDetailMenuProps & TaskDetailMenuActions;
+
+export const TaskDetailMenuComponent: React.StatelessComponent<Props> = (props: Props): JSX.Element => (
+    <FloatingActionButtonsComponent openIcon={'close'} closedIcon={'th-list'} buttons={getTaskDetailMenuButtons(props)} />
+);
+
+const getTaskDetailMenuButtons = (props: Props): ReadonlyArray<JSX.Element> => {
+    return R.map((button: TaskStateButton) => {
+        switch (button) {
+            default:
```

If we reach this switch with a value that's different from the cases below, that is actually a bug. Having default fall through to the next case hides that kind of bug. It would actually be better to throw an exception here.

---

Fixed